ca/server.go

```diff
@@ -460,80 +490,43 @@ func (s *Server) updateCluster(ctx context.Context, cluster *api.Cluster) {
 			}).WithError(err).Warn("failed to parse certificate expiration, using default")
 
 		}
-		rCA := cluster.RootCA
+		// Attempt to update our local RootCA with the new parameters
 		err = s.securityConfig.UpdateRootCA(rCA.CACert, rCA.CAKey, expiry)
 		if err != nil {
 			log.G(ctx).WithFields(logrus.Fields{
 				"cluster.id": cluster.ID,
 				"method":     "(*Server).updateCluster",
-			}).WithError(err).Error("updating root key failed")
+			}).WithError(err).Error("updating Root CA failed")
 		} else {
 			log.G(ctx).WithFields(logrus.Fields{
 				"cluster.id": cluster.ID,
 				"method":     "(*Server).updateCluster",
-			}).Debugf("root CA updated successfully")
+			}).Debugf("Root CA updated successfully")
 		}
 	}
 }
 
-func (s *Server) setNodeCertState(node *api.Node, state api.IssuanceStatus_State) error {
-	return s.store.Update(func(tx store.Tx) error {
-		latestNode := store.GetNode(tx, node.ID)
-		if latestNode == nil {
-			return store.ErrNotExist
-		}
-
-		// Remote users are expecting a full certificate chain, not just a signed certificate
-		latestNode.Certificate.Status = api.IssuanceStatus{
-			State: state,
-		}
-
-		return store.UpdateNode(tx, latestNode)
-	})
-}
-
+// evaluateAndSignNodeCert implements the logic of which certificates to sign
 func (s *Server) evaluateAndSignNodeCert(ctx context.Context, node *api.Node) {
 	// If the desired membership and actual state are in sync, there's
 	// nothing to do.
 	if node.Spec.Membership == api.NodeMembershipAccepted && node.Certificate.Status.State == api.IssuanceStateIssued {
 		return
 	}
-	if node.Spec.Membership == api.NodeMembershipRejected && node.Certificate.Status.State == api.IssuanceStateRejected {
-		return
-	}
-
-	// If the desired membership was set to rejected, we should
-	// act on that right away, and that is all that should be done.
-	if node.Spec.Membership == api.NodeMembershipRejected {
-		err := s.setNodeCertState(node, api.IssuanceStateRejected)
-		if err != nil {
-			log.G(ctx).WithFields(logrus.Fields{
-				"node.id":   node.ID,
-				"node.role": node.Certificate.Role,
-				"method":    "(*Server).evaluateAndSignCert",
-			}).WithError(err).Errorf("failed to change certificate state")
-		}
-		return
-	}
 
 	// If the certificate state is renew, then it is a server-sided accepted cert (cert renewals)
 	if node.Certificate.Status.State == api.IssuanceStateRenew {
 		s.signNodeCert(ctx, node)
 		return
 	}
 
-	// If the certificate state is not pending at this point, we are in an unknown state, return
-	if node.Certificate.Status.State != api.IssuanceStatePending {
-		return
-	}
```

Comment:
I'm not sure about removing this. Won't it cause us to enter an infinite signing loop if State == FAILED?